src/explorer/query_data.rs

```diff
@@ -0,0 +1,920 @@
+        }
+    }
+}
+
+/// [GenesisOverview] provides a summary overview of the block chain since
+/// it's genesis. At a high level it includes the total number of unique
+/// rollups, transactions, and blocks that are in the block chain.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GenesisOverview {
+    pub rollups: u64,
+    pub transactions: u64,
+    pub blocks: u64,
+    // pub sequencer_nodes: u64,
+}
+
+/// [ExplorerHistograms] provides a series of data points that can be used to
+/// draw simple histograms for the Block Explorer.  The data returned is meant
+/// to be an optimal packing of the values being returned.
+///
+/// It contains data for the last N blocks, indicated by the length of the
+/// vectors contained within the struct.  All of the vectors **MUST** have the
+/// same length.  The labels of the graph points is the `block_heights` vector.
+/// The remaining data points are the `block_time`, `block_size`, and
+/// `block_transactions` for those `block_heights`.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ExplorerHistograms {
+    pub block_time: Vec<u64>,
+    pub block_size: Vec<u64>,
+    pub block_transactions: Vec<u64>,
+    pub block_heights: Vec<u64>,
+}
+
+/// [ExplorerSummary] is a struct that represents an at-a-glance snapshot of
+/// the Block Chain.  It contains some helpful information that can be used
+/// to display a simple health check of the Block Chain.
+///
+/// It contains the latest block for reference, the most recent blocks, the
+/// most recent transactions, some statistics concerning the total number
+/// of elements contained within the chain, and some histograms that can be
+/// used to draw graphs for the Block Explorer.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct ExplorerSummary<Types: NodeType>
+where
+    Header<Types>: ExplorerHeader<Types>,
+{
+    pub latest_block: BlockDetail<Types>,
+    pub genesis_overview: GenesisOverview,
+    pub latest_blocks: Vec<BlockSummary<Types>>,
+    pub latest_transactions: Vec<TransactionSummary<Types>>,
+    //  Most Active Rollups
+    pub histograms: ExplorerHistograms,
+}
+
+/// [SearchResult] is a struct that represents the results of executing a
+/// search query against the chain.  It contains a list of blocks and
+/// transactions that match the search query.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct SearchResult<Types: NodeType>
+where
+    Header<Types>: ExplorerHeader<Types>,
+{
+    pub blocks: Vec<BlockSummary<Types>>,
+    pub transactions: Vec<TransactionSummary<Types>>,
+}
+
+/// [GetBlockDetailError] represents an error that has occurred in response to
+/// the [GetBlockDetail] request.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+
+pub enum GetBlockDetailError {
+    #[serde(untagged)]
+    Unimplemented(Unimplemented),
+    #[serde(untagged)]
+    BlockNotFound(String),
+    #[serde(untagged)]
+    QueryError(QueryError),
+}
```

Comment:
```suggestion
  #[serde(untagged)]
pub enum GetBlockDetailError {
    Unimplemented(Unimplemented),
    BlockNotFound(String),
    QueryError(QueryError),
}
```

Reply:
This has been adjusted in commit `c7dedab5219c8fa78f3cb7cac872cd2d2412ccb7`.